presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -153,4 +215,77 @@ static long timeUnitsToMillis(String timeWithUnits)
     {
         return Duration.valueOf(timeWithUnits).toMillis();
     }
+
+    private static class EffectiveLimits
+    {
+        private ZonedDateTime now;
+        private String maxQueryRangeDurationStr;
+        private Optional<PrometheusPredicateTimeInfo> maybePredicateRange;
+        private ZonedDateTime nowOrUpperBound;
+        private java.time.Duration maxQueryRangeDuration;
+
+        public EffectiveLimits(LocalDateTime now, String maxQueryRangeDurationStr, Optional<PrometheusPredicateTimeInfo> maybePredicateRange)
+        {
+            this.now = ZonedDateTime.of(now, ZoneId.systemDefault());
+            this.maxQueryRangeDurationStr = maxQueryRangeDurationStr;
+            this.maybePredicateRange = maybePredicateRange;
+        }
+
+        public ZonedDateTime getNowOrUpperBound()
+        {
+            return nowOrUpperBound;
+        }
+
+        public java.time.Duration getMaxQueryRangeDuration()
+        {
+            return maxQueryRangeDuration;
+        }
+
+        public EffectiveLimits invoke()
```

Comment:
Any reason this logic doesn't just happen in the constructor? I'm not sure if there's a need for a separate method to initialize the state.

Reply:
Yeah was able to simplify it a bunch in ef93727